client/signup/steps/site-type/index.jsx

```diff
@@ -0,0 +1,207 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import i18n, { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import StepWrapper from 'signup/step-wrapper';
+import SignupActions from 'lib/signup/actions';
+import formState from 'lib/form-state';
+import { setSiteType } from 'state/signup/steps/site-type/actions';
+import { getSiteType } from 'state/signup/steps/site-type/selectors';
+
+//Form components
+import Card from 'components/card';
+import Button from 'components/button';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormRadio from 'components/forms/form-radio';
+
+class SiteType extends Component {
+	constructor( props ) {
+		super( props );
+		this._isMounted = false;
+		this.state = {
+			siteType: props.siteType,
+		};
+	}
+
+	componentDidMount() {
+		this._isMounted = true;
+		this.formStateController = new formState.Controller( {
+			fieldNames: [ 'siteType' ],
+			validatorFunction: noop,
+			onNewState: this.setFormState,
+			hideFieldErrorsOnChange: true,
+			initialState: {
+				siteType: {
+					value: this.props.siteType,
+				},
+			},
+		} );
+		this.setFormState( this.formStateController.getInitialState() );
+	}
+
+	componentWillUnmount() {
+		this._isMounted = false;
+	}
+
+	setFormState = state => {
+		this._isMounted && this.setState( { form: state } );
+	};
+
+	handleRadioChange = event => {
+		this.formStateController.handleFieldChange( {
```

Comment:
Internal state (`this.state`) should be enough. Unless there's a real need for validation or complex form control state management, we could probably revert all this.